elements/pl-big-o-input/pl-big-o-input.py

```diff
@@ -0,0 +1,317 @@
+    variables = phs.get_variables_list(
+        pl.get_string_attrib(element, "variables", VARIABLES_DEFAULT)
+    )
+
+    if len(variables) > 1:
+        raise ValueError(f"Only one variable is supported for question {name}")
+
+    if pl.has_attrib(element, "correct-answer"):
+        if name in data["correct_answers"]:
+            raise ValueError(f"duplicate correct_answers variable name: {name}")
+
+        a_true = pl.get_string_attrib(element, "correct-answer")
+        # Try converting answer before storing
+        try:
+            phs.convert_string_to_sympy(
+                a_true, variables, allow_complex=False, allow_trig_functions=False
+            )
+        except phs.BaseSympyError:
+            raise ValueError(f"Parsing correct answer {a_true} for {name} failed")
+
+        data["correct_answers"][name] = a_true
+
+
+def render(element_html: str, data: pl.QuestionData) -> str:
+    element = lxml.html.fragment_fromstring(element_html)
+    name = pl.get_string_attrib(element, "answers-name")
+    variables = phs.get_variables_list(
+        pl.get_string_attrib(element, "variables", VARIABLES_DEFAULT)
+    )
+    display = pl.get_enum_attrib(pl.DisplayType, element, "display", DISPLAY_DEFAULT)
+    size = pl.get_integer_attrib(element, "size", SIZE_DEFAULT)
+
+    bigo_type = pl.get_enum_attrib(BigOType, element, "type", BIG_O_TYPE_DEFAULT).value
+
+    operators: List[str] = [
+        "exp",
+        "log",
+        "sqrt",
+        "factorial",
+        "( )",
+        "+",
+        "-",
+        "*",
+        "/",
+        "^",
+        "**",
+    ]
+    constants: List[str] = ["pi", "e"]
+
+    info_params = {
+        "format": True,
+        "variables": variables,
+        "operators": operators,
+        "constants": constants,
+    }
+
+    PARTIAL_SCORE_DEFAULT: pl.PartialScore = {"score": None}
+    score = data["partial_scores"].get(name, PARTIAL_SCORE_DEFAULT).get("score")
+
+    if data["panel"] == "question":
+        editable = data["editable"]
+        raw_submitted_answer = data["raw_submitted_answers"].get(name)
+
+        with open("pl-big-o-input.mustache", "r", encoding="utf-8") as f:
+            info = chevron.render(f, info_params).strip()
+
+        if raw_submitted_answer is not None:
+            raw_submitted_answer = escape(raw_submitted_answer)
+
+        score_type, score_value = pl.determine_score_params(score)
+
+        html_params = {
+            "question": True,
+            "name": name,
+            "editable": editable,
+            "info": info,
+            "size": size,
+            "show_info": pl.get_boolean_attrib(
+                element, "show-help-text", SHOW_HELP_TEXT_DEFAULT
+            ),
+            "uuid": pl.get_uuid(),
+            display.value: True,
+            "show_placeholder": size >= PLACEHOLDER_TEXT_THRESHOLD,
+            "raw_submitted_answer": raw_submitted_answer,
+            "type": bigo_type,
+            score_type: True,
```

Comment:
👀 What's this? Ditto for the submission panel below.

Reply:
Setting the score type? Basically, that's getting the output of the corresponding helper function `determine_score_params` and adding it to the display dictionary. It's serving the same purpose as the code block below (that appears duplicated in the code for basically every element):

![image](https://user-images.githubusercontent.com/1345068/206319011-ce5d5870-f7b2-4a20-aba9-1913655abecc.png)

When writing this, we thought it was a good idea to consolidate this logic. As an aside, the exception catch and re-throw in the above block is unnecessary. We think that using this helper function can really be used by every element to replace the block above (since that just contributes to code bloat).

Reply:
Oh man, sorry, this is my JavaScript brain not differentiating between `"object_key"` and `object_key`. I see what's happening now, there's not a literal `"score_type"` key in the dict.